src/main/java/com/opentok/OpenTok.java

```diff
@@ -33,427 +33,506 @@
-     */
-    public Archive stopArchive(String archiveId) throws OpenTokException {
-
-        String archive = this.client.stopArchive(archiveId);
-        try {
-            return archiveReader.readValue(archive);
-        } catch (Exception e) {
-            throw new RequestException("Exception mapping json: " + e.getMessage());
-        }
-    }
-    
-    /**
-     * Deletes an OpenTok archive.
-     * <p>
-     * You can only delete an archive which has a status of "available" or "uploaded". Deleting an
-     * archive removes its record from the list of archives. For an "available" archive, it also
-     * removes the archive file, making it unavailable for download.
-     *
-     * @param archiveId The archive ID of the archive you want to delete.
-     */
-    public void deleteArchive(String archiveId) throws OpenTokException {
-        this.client.deleteArchive(archiveId);
-    }
+		return startArchive(sessionId, new ArchiveProperties.Builder().build());
+	}
+
+	public Archive startArchive(String sessionId, String name) throws OpenTokException {
+		ArchiveProperties properties = new ArchiveProperties.Builder().name(name).build();
+		return startArchive(sessionId, properties);
+	}
+
+	/**
+	 * Stops an OpenTok archive that is being recorded.
+	 * <p>
+	 * Archives automatically stop recording after 120 minutes or when all
+	 * clients have disconnected from the session being archived.
+	 *
+	 * @param archiveId
+	 *            The archive ID of the archive you want to stop recording.
+	 * @return The Archive object corresponding to the archive being stopped.
+	 */
+	public Archive stopArchive(String archiveId) throws OpenTokException {
+
+		String archive = this.client.stopArchive(archiveId);
+		try {
+			return archiveReader.readValue(archive);
+		} catch (Exception e) {
+			throw new RequestException("Exception mapping json: " + e.getMessage());
+		}
+	}
+
+	/**
+	 * Deletes an OpenTok archive.
+	 * <p>
+	 * You can only delete an archive which has a status of "available" or
+	 * "uploaded". Deleting an archive removes its record from the list of
+	 * archives. For an "available" archive, it also removes the archive file,
+	 * making it unavailable for download.
+	 *
+	 * @param archiveId
+	 *            The archive ID of the archive you want to delete.
+	 */
+	public void deleteArchive(String archiveId) throws OpenTokException {
+		this.client.deleteArchive(archiveId);
+	}
+
+	public static class Builder {
+		private int apiKey;
+		private String apiSecret;
+		private String apiUrl;
+		private Proxy proxy;
+
+		
+		public Builder(int apiKey, String apiSecret) {
+			this.apiKey = apiKey;
+			this.apiSecret = apiSecret;
+		}
+
+		public Builder apiUrl(String apiUrl) {
+			this.apiUrl = apiUrl;
+			return this;
+		}
+
+		public Builder proxy(Proxy proxy) {
+			this.proxy = proxy;
+			return this;
+		}
+
+		public OpenTok build() {
+			if (this.apiUrl != null && this.proxy != null) {
```

Comment:
This `if` statement creates a lot of duplication.

You could eliminate the duplication by doing something like:

```
HttpClient.Builder clientBuilder = new HttpClient.Builder(this.apiKey, this.apiSecret);
if (this.apiUrl != null) clientBuilder.apiUrl(this.apiUrl);
if (this.proxy != null) clientBuilder.proxy(this.proxy);
return new OpenTok(this.apiKey, this.apiSecret, clientBuilder.build()); // this line requires a new private constructor that takes an HttpClient parameter
```